pyadjoint/overloaded_type.py

```diff
@@ -320,6 +320,28 @@ def _ad_str(self):
         """
         return str(self)
 
+    def _ad_petsc_vec_read_only(self):
+        """This method must be overridden.
```

Comment:
I think that in this case this overstates the case.
```suggestion
        """This method should be overwritten by types which can return a PETSc Vec.
```